### ISSUE TITLE
ordering of legend labels

### ISSUE THREAD

**Issue by Alcampopiano** [altair-viz/altair#1059]
Hi Jake,

Can we change around the order of the legend entries?

Say, put Google at the top in this example from the docs:

```python
import altair as alt
from vega_datasets import data

stocks = data.stocks()

alt.Chart(stocks).mark_line().encode(
    x='date',
    y='price',
    color='symbol'
)
``` 
![visualization 12](https://user-images.githubusercontent.com/4624246/43413693-be70d1b4-93fe-11e8-840a-9876762e8f93.png)

**Comment by jakevdp** [comment]
You can specify the order using the ``sort`` property of the channel. Categories will appear in the specified order, and any remaining categories will be appended alphabetically. So if all you care is that GOOG is first, you can do this:
```python
import altair as alt
from vega_datasets import data

stocks = data.stocks()

alt.Chart(stocks).mark_line().encode(
    x='date',
    y='price',
    color=alt.Color('symbol', sort=['GOOG'])
)
```
![visualization 29](https://user-images.githubusercontent.com/781659/43413795-e1a91894-93e5-11e8-868f-250d00db5d38.png)

**Comment by Alcampopiano** [comment]
Oh this is great. I should have guessed that. Thanks very much.

**Comment by Alcampopiano** [comment]
Hi Jake,

I'm trying to follow your instructions here for reording the legend labels. For some reason the "stock prices" example works, but when using my own data I get an error despite the same data structure and Altair code. What am I missing here?

### this works
```python
import altair as alt
from vega_datasets import data

stocks = data.stocks()

alt.Chart(stocks).mark_line().encode(
    x='date',
    y='price',
    color=alt.Color('symbol', sort=['GOOG', 'IBM'])
)
```

### this doesn't
![df](https://user-images.githubusercontent.com/4624246/43589059-e64e63ce-963b-11e8-98f8-4e67f4e29c22.png)

(I'm hiding some of the data here but you can see the structure)
```python
alt.Chart(data_melt).mark_line().encode(
    x='year',
    y='percentage of students',
    color=alt.Color('percent categories', sort=['applied to university only', 'applied to college only'])
)
```

**Comment by jakevdp** [comment]
I can't reproduce this error. If I run the following it works without an issue:
```python
import altair as alt
import pandas as pd

data = pd.DataFrame({'year': range(2010, 2016),
                     'percentage of students': range(10, 16),
                     'percent categories': 3 * ['applied to university only', 'applied to college only']})

alt.Chart(data).mark_line().encode(
    x='year',
    y='percentage of students',
    color=alt.Color('percent categories', sort=['applied to university only', 'applied to college only'])
)
```

**Comment by jakevdp** [comment]
This error is not coming from Python, it is coming from Javascript, so it is the javascript library version that is relevant here. That library version is not controlled by Altair, but controlled by the extension your frontend (jupyterlab, jupyter notebook, colab, etc.) is using to render Altair's outputs.

**Comment by Alcampopiano** [comment]
Okay thanks. I will upgrade JL and try again.

**Comment by Alcampopiano** [comment]
I updated JL and your example works but only if no layering present. If I layer some dots, the legend goes back to its original ordering.

```python

import altair as alt
import pandas as pd

data = pd.DataFrame({'year': range(2010, 2016),
                     'percentage of students': range(10, 16),
                     'percent categories': 3 * ['applied to university only', 'applied to college only']})


base=alt.Chart(data).encode(    
    x='year:O',
    y='percentage of students',
    color=alt.Color('percent categories', sort=['applied to university only', 'applied to college only'])
)

line=base.mark_line()
dot=base.mark_circle()
both=line+dot
both
```

![visualization 13](https://user-images.githubusercontent.com/4624246/43591076-7e665cbc-9640-11e8-971b-6ecd89364ffe.png)

**Comment by jakevdp** [comment]
That is a known issue in Vega-Lite: see https://github.com/vega/vega-lite/issues/2177 and https://github.com/altair-viz/altair/issues/820

**Comment by Alcampopiano** [comment]
Ah I see. Thanks for letting me know.

**Comment by jakevdp** [comment]
This should be fixed as of Altair 4.0
